File: src/2025-03-04-12-20-21--deepseek-vr-0304/python/exercises/practice/grade-school/grade_school.py
class School:
    def __init__(self):
        self.students = {}
        self.all_students = set()
        self.added_results = []

    def add_student(self, name, grade):
        if name in self.all_students:
            self.added_results.append(False)
            return
        if grade not in self.students:
            self.students[grade] = []
        self.students[grade].append(name)
        self.students[grade].sort()
        self.all_students.add(name)
        self.added_results.append(True)

    def roster(self):
        roster_list = []
        for grade in sorted(self.students.keys()):
            roster_list.extend(self.students[grade])
        return roster_list

    def grade(self, grade_number):
        return self.students.get(grade_number, []).copy()

    def added(self):
        return self.added_results

File: src/2025-03-04-12-20-21--deepseek-vr-0304/cpp/exercises/practice/grade-school/grade_school.cpp
#include "grade_school.h"

namespace grade_school {

bool school::add(const std::string& name, int grade) {
    auto& students = roster_[grade];
    return students.insert(name).second;
}

std::vector<std::string> school::grade(int grade) const {
    auto it = roster_.find(grade);
    if (it != roster_.end()) {
        return {it->second.begin(), it->second.end()};
    }
    return {};
}

std::map<int, std::vector<std::string>> school::roster() const {
    std::map<int, std::vector<std::string>> result;
    for (const auto& [gr, names] : roster_) {
        result[gr] = {names.begin(), names.end()};
    }
    return result;
}

}  // namespace grade_school
